complete/src/main/java/com/example/restservice/controller/advices/AuthenticationFilter.java

```diff
@@ -10,31 +10,37 @@
 import com.auth0.jwt.interfaces.DecodedJWT;
 import com.example.restservice.exceptions.SQAccessDeniedException;
 import java.io.IOException;
-import java.lang.reflect.Type;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.security.interfaces.RSAPublicKey;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.core.MethodParameter;
-import org.springframework.http.HttpInputMessage;
-import org.springframework.http.converter.HttpMessageConverter;
-import org.springframework.web.bind.annotation.ControllerAdvice;
-import org.springframework.web.servlet.mvc.method.annotation.RequestBodyAdviceAdapter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
 
-@ControllerAdvice
-public class SecurityAdvice extends RequestBodyAdviceAdapter {
+
+@Component
+@Order(1)
+public class AuthenticationFilter implements Filter {
```

Comment:
Auth stuff should be done as a Filter. Advices are not the right place to put this. RequestBodyAdvise would, for example, get triggered only on requests with a body.

Once we have a API Gateway set up, I'm hoping we can do auth at API gateway level too, but this is working.